enhancements/machine-api/cluster-autoscaler-operator.md

```diff
@@ -0,0 +1,353 @@
+---
+title: cluster-autoscaler-operator
+authors:
+  - "@JoelSpeed"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-05-19
+last-updated: 2020-05-19
+status: implemented
+see-also:
+replaces:
+superseded-by:
+---
+
+# Cluster Autoscaler Operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Provide a declarative way for users to enable autoscaling of compute resources by providing the Cluster Autoscaler
+as a Deployment with opinionated default options.
+
+Provide a way for users to opt individual MachineSets into autoscaling and to set boundaries for the size of the
+MachineSets that they wish to autoscale.
+
+## Motivation
+
+### Goals
+
+- Enable users to create Cluster Autoscaler with opinionated defaults
```

Comment:
remove the operational burden for users to get autoscaling up and running quickly?